### ISSUE TITLE
Plugins loaded in Alphabetical order.

### ISSUE THREAD

**Issue by Psyonics** [Solero/Houdini#15]
Not as much an issue but more of an annoyance. You see because the plugins load in alphabetical order, if you have a plugin that depends on another plugin with a higher priority in terms of alphabetical order than the plugin it is depending on, you will get a error due to the plugin that is being depended on not being found, because the plugin that depends on it is loaded first. Not exactly sure if anyone else has had this happen to them, might just be me, but I thought that posting this would be an alright idea nevertheless.

**Comment by ketnipz** [comment]
What would you suggest is a good solution for this? I guess all plugins could have an optional "info" file which lists dependant plugins. Or something along these lines.

**Comment by valid22** [comment]
Dependent like how? I don't think what you say could cause error unless the plug-in your are trying to import doesn't exist

**Comment by ketnipz** [comment]
Some plugins in Houdini can depend on other plugins, by getting their instance like:

`self.bot = self.server.plugins["Bot"]`

Also: a solution to this would just to reference the dependancy plugin in the dependant plugin's `ready` method, no?

```
def ready(self):
    self.bot = self.server.plugins["Bot"]
```

`ready()` is invoked once ALL plugins are instantiated.

**Comment by valid22** [comment]
Fair enough. The way I got over this is by importing all plugins once, then creating instances of each plug-in class by taking account of their dependent plugins first (then call they method in plug-in which has the plug-in it depends on as arguments)

So importing all plugins first should do it, then create instances of it
